keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1458 @@
+
+#### Scheduler Changes
+
+The scheduler will determine pod resources requirements in the following ways:
+1. Directly from Pod-Level Resources (pod.spec.resources)
+   
+   If the pod has pod.spec.resources defined, the scheduler will use these values (requests and limits) as the sole indicator of the pod's resource needs, including init containers, sidecar containers and regular containers.
+2. Indirectly from Container-level Resources
+   
+   If the pod does NOT have pod.spec.resources defined, the scheduler will derive the pod's resource needs by aggregating the requests and limits of all containers (init, sidecar and regular containers) within the pod.
+3. Both Pod-level and Container level
+   
+   If both pod-level and container-level resources are specified, the scheduler will prioritize the pod-level resources and ignore the container-level requests for node placement decisions.
+
+#### QoS Changes
+
+QoS is determined by derived Pod Resources which considers (preferred in order):
+1. The values for resources specified in the pod’s `resources` stanza.
+2. The value of resources specified in the container’s `resources` stanza.
+
+In simple words, if `resources` is specified at pod-level, the values in pod-level `resources` stanza are given a preference in determining the QoS class, otherwise container level `resources` are considerted for QoS determination.
+
+* **Guaranteed QoS**
+
+    A pod is considered `Guaranteed`, if for each resource type, either of following is true:
+    * It is explicitly resourced (pod.spec.resources is set) for that resource type AND the pod's request is equal to its limits.
+    * It is implicitly resourced (pod.spec.resources is not set) AND all containers requests are equal to their limits.
+
+  **Note: If the pod spec includes `resources` stanza at pod level, but either requests or limits are missing for a resource, the pod will not be considered Guaranteed QoS, even if container-level requests and limits are present and are equal.**
+
+* **Burstable QoS Criteria**
+
+    A pod is considered `Burstable`, if for each resource type it doesn't meet the criteria for Guaranteed QoS for that resource, and either of following is true:
+
+    * It is explicitly resourced (pod.spec.resources is set) for that resource, AND request (less than or equal to limit) or limit is set.
+    * It is implicitly resourced for that resource AND for at least one container in the pod, request (less than or equal to limit) or limit is set. 
+  
+
+* **BestEffort QoS Criteria**
+
+    None of pod-level or container-level resources requests or limits are specified.
+
+
+#### OOM Killer Behavior
+With Pod level resources, the OOM Killer will evolve to consider the cumulative resource limits set for the entire Pod. If the aggregated
+memory usage of all the containers within a Pod exceeds the effective pod level limits, the entire Pod becomes a candidate for OOM Killing, even if individual containers are within their limits.
```

Comment:
Do you mean we will try to OOM kill the whole pod like we did for all processes inside the container: https://github.com/kubernetes/kubernetes/pull/117793?